### DISCUSSION TITLE
can't send email

### DISCUSSION THREAD

**Discussion by fav22** [postalserver/postal#1739] [Help with using Postal]
Hello , when i try to send email test , it has always blocked on queue ! can someone here can tell me what's the issue please ? Thank you all

**Comment by ghost** [comment]
Can you provide some screenshots of the server's page?

**Reply by fav22** [reply]
did you mean mail server or vps server page please ?

**Comment by ghost** [comment]
The mail server, on the "overview" page

**Comment by fav22** [comment]


**Reply by ghost** [reply]
Can you send a screenshot of what it looks like when you click on one of those pending emails?

**Reply by fav22** [reply]
![Uploading like.PNG…]()

**Reply by ghost** [reply]
That image doesn't load for me

**Reply by fav22** [reply]
sorry , i'll resend it

**Reply by ghost** [reply]
What do you get when you run `postal status` on your server?

**Reply by fav22** [reply]
`postal_cron_1       /docker-entrypoint.sh post ...   Up
postal_requeuer_1   /docker-entrypoint.sh post ...   Exit 1
postal_smtp_1       /docker-entrypoint.sh post ...   Up
postal_web_1        /docker-entrypoint.sh post ...   Up
postal_worker_1     /docker-entrypoint.sh post ...   Exit 1
root@postal:~#
`

**Reply by fav22** [reply]
![status](https://user-images.githubusercontent.com/95019796/143683816-3afbb9f1-c19e-44dd-a18f-483aa90d2a47.PNG)

**Reply by fav22** [reply]
what do you say about it ?

**Reply by ghost** [reply]
The requeuer and worker processes being down probably explains why the message queue isn't working. I'm not too familiar with how Postal works internally, but I think it relies on RabbitMQ. Did you set that up correctly? It explains how to in the docs. Send me the output of `docker ps`, that shows which docker containers are running.

**Reply by fav22** [reply]
here please
![docker](https://user-images.githubusercontent.com/95019796/143684792-e8c0e935-08ca-4c61-8c0d-49433254cf3b.PNG)

**Reply by ghost** [reply]
Where are you running mysql/mariadb?

**Reply by fav22** [reply]
when i tried to install Rabbitmq , there was this error
![rabbit](https://user-images.githubusercontent.com/95019796/143684937-008320ee-e5e4-4ed8-991a-fe7ba77eb88f.PNG)

**Reply by fav22** [reply]
i execute mysql and mariadb by ssh

**Reply by ghost** [reply]
> when i tried to install Rabbitmq , there was this error ![rabbit](https://user-images.githubusercontent.com/95019796/143684937-008320ee-e5e4-4ed8-991a-fe7ba77eb88f.PNG)

Yeah, you probably already have rabbitmq running then. Try running `docker container logs 380` and send the logs of RabbitMQ.

**Reply by fav22** [reply]
it has makes an error , a bug 
![ebug](https://user-images.githubusercontent.com/95019796/143685316-fb56a14f-87c0-4718-af81-d85b5e039d37.PNG)

**Reply by fav22** [reply]
how can i do to restart or delet rabbitmq on my system so i think i'll be able to install it safely

**Reply by ghost** [reply]
You probably set it up incorrectly, so I'd recommend removing the rabbitmq container and doing it again.

1. Stop postal with `postal stop`
2. Stop the rabbitmq container with `docker container stop 380`
3. Remove the container with `docker container rm 380`
4. Generate a random password. I usually make it 128 characters long. Put it in notepad or something temporarily so you don't lose it.
5. Start rabbitmq again with `docker run -d --name postal-rabbitmq -p 127.0.0.1:5672:5672 --restart always -e RABBITMQ_DEFAULT_USER=postal -e RABBITMQ_DEFAULT_PASS=PUT_YOUR_PASSWORD_HERE -e RABBITMQ_DEFAULT_VHOST=postal rabbitmq:3.8`. Obviously replace `PUT_YOUR_PASSWORD_HERE` with the password you just made.
6. Open `/opt/postal/config/postal.yml` and go down to the "rabbitmq" section, then remove the password there that you've already set and change it to your new one.
7. Start postal again with `postal start`

Let me know if it works then, and run the stuff I sent before to check the status and logs :)

**Reply by fav22** [reply]
unfortunately , nothing has changed

**Reply by fav22** [reply]
Resolu , merci l'equipe

**Reply by ghost** [reply]
How did you fix it?

**Reply by willpower232** [reply]
As mentioned in other discussions, rabbit mq can be slower to start from a fresh boot than Postal so if they're all on the same server then you need to configure those two bits of Postal to wait for rabbit.

**Reply by fav22** [reply]
> How did you fix it?

simply by reconfiguring correctly `postal.yml`
